pkg/resources/api_integration.go

```diff
@@ -77,6 +77,12 @@ var apiIntegrationSchema = map[string]*schema.Schema{
 		Optional:    true,
 		Description: "Lists the endpoints and resources in the HTTPS proxy service that are not allowed to be called from Snowflake.",
 	},
+	"api_key": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Default:     "",
```

Comment:
if this is optional you don't need to set a default value to an empty string. instead you can rely on the `d.GetOk()` function instead.
```
if _, ok := d.GetOk("api_key);ok {
 // api_key was set
} 
// else no need to do anything
```

Reply:
@sfc-gh-swinkler Great catch, I made the changes. Thanks for reviewing